### PR TITLE
Allow case insensitive search for `.tmtheme` paths

### PR DESCRIPTION
**Pull request by gjtorikian**
This was an annoying bug to unravel. 😓 On case insensitive systems, like macOS, `tmtheme` ought to be just as valid as `tmTheme`.

### CODE REVIEW

**Comment by gjtorikian** [comment]
Is this project still maintained?

**Comment by Enselic** [comment]
Several people are watching project activity as far I know. The main reason no action has been taken on this PR is probably that CI does not pass. At least that's the main reason for me personally.

**Comment by keith-hall** [comment]
Indeed, I wanted to help investigate the CI failures but forgot and haven't had time.

I would probably also expect it to detect the filesystem case sensitivity and act accordingly, not just always assume insensitive is okay, otherwise this change could break it for others, no?

**Comment by gjtorikian** [comment]
> The main reason no action has been taken on this PR is probably that CI does not pass. At least that's the main reason for me personally.

Sorry, [this comment](https://github.com/trishume/syntect/pull/467#issuecomment-1450625734) was meant to ask for help on that. My previous PR with CI passing (https://github.com/trishume/syntect/pull/466) also hasn't been merged, so I wasn't sure if anyone was looking here.

> I would probably also expect it to detect the filesystem case sensitivity and act accordingly, not just always assume insensitive is okay

There are two other locations in the existing code where case insensitivity is being used:
https://github.com/trishume/syntect/blob/c61ce60c72d67ad4e3dd06d60ff3b13ef4d2698c/src/parsing/syntax_set.rs#L186-L200

In other words, this is the only code location where case sensitivity is (incorrectly, imho) required.

**Comment by keith-hall** [comment]
`sublime-syntax` files are looked for case sensitively: https://github.com/trishume/syntect/blob/master/src/parsing/syntax_set.rs#L490

As for file extensions when looking up a syntax, it has been proven that Sublime Text does this in a case insensitive manner, and looking up a syntax for an arbitrary token is just a convenience that syntect offers, and it makes sense to be insensitive.

So if we can prove that Sublime also handles varying case `tmTheme` and `sublime-syntax` files on a case-sensitive filesystem, then it would make sense to add a test for it and then, yes, merge it. Or if it handles it only on a case-insensitive filesystem, then we should do the same in syntect. But until then, I feel we could be breaking Sublime compatibility with this change...

**Comment by keith-hall** [comment]
I was able to reproduce this failing test locally. It seems it is because there is a folder called `InspiredGitHub.tmtheme`.  I suggest we change the code to ensure we only look for files.

**Comment by keith-hall** [comment]
And for what it's worth, I tested Sublime Text 4148 on Linux (i.e. case-sensitive file system) and `.Sublime-Syntax` files are not loaded, and `.tmtheme` files don't show up in the color scheme picker UI. So we just need to prove how it works on a case-insensitive file system to decide the way forward here.

**Comment by gjtorikian** [comment]
> So we just need to prove how it works on a case-insensitive file system to decide the way forward here.

I am not sure if this is a task for me or the repo's administrators, but, Sublime 4143 on macOS (case-insensitive) will happily load a file whether it's `.tmTheme` or `.tmtheme`.

**Comment by keith-hall** [comment]
Cool, so let's fix it to only look for files and you will get my approval :)

**Comment by gjtorikian** [comment]
Done!

**Review by keith-hall — APPROVED** [review]

**Comment by Enselic** [comment]
Thank you for the contribution.

Indeed it is mostly formatting changes. For the record, I am strongly in favor of syntect running cargo fmt as part of CI. That way the problem of PRs containging pure formatting changes goes away completely. It is a very common practice in the Rust ecosystem (and other ecosystems) in my experience, and it works great. It is a time saver both for maintainers and contributors.

**Review by mikelnight83 — COMMENTED** [review]
[
Screenshot (1)
](url)
